sixtyfps_compiler/passes/lower_layout.rs

```diff
@@ -502,7 +504,7 @@ fn eval_const_expr(
 ) -> Option<u16> {
     match expression {
         Expression::NumberLiteral(v, Unit::None) => {
-            if *v < 0. || *v > u16::MAX as f64 || v.trunc() != *v {
+            if *v < 0. || *v > u16::MAX as f64 || !v.trunc().approx_eq(v) {
```

Comment:
I've seen this one before. Hasn't this already been done? Or is that duplicate code?

Reply:
Commit 2cdc5848a2b77a52b4c8611410fbfb79c3743ea7 changed this 

> I changed the code to decide whether a number is a positive integer to
> make do without a comparison along the way.

```patch
diff --git a/sixtyfps_compiler/passes/lower_layout.rs b/sixtyfps_compiler/passes/lower_layout.rs
index 6d538546..92b94475 100644
--- a/sixtyfps_compiler/passes/lower_layout.rs
+++ b/sixtyfps_compiler/passes/lower_layout.rs
@@ -493,12 +493,11 @@ fn eval_const_expr(
 ) -> Option<u16> {
     match expression {
         Expression::NumberLiteral(v, Unit::None) => {
-            let r = *v as u16;
-            if (r as f32 - *v as f32).abs() > f32::EPSILON {
+            if *v < 0. || *v > u16::MAX as f64 || v.trunc() != *v {
                 diag.push_error(format!("'{}' must be a positive integer", name), span);
                 None
```

and accidentally introduced this comparison :).